[NEW] GitHub Component Kit

samad-yar-khan opened the pull request:
-  [x] I have read the Contributing Guide - https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat doc
  -  [x] I have signed the CLA - https://cla-assistant.io/RocketChat/Rocket.Chat
  - [x] Lint and unit tests pass locally with my changes
  - [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
  -  [x] I have added necessary documentation (if applicable)
  - [x] Any dependent changes have been merged and published in downstream modules

## Issues

Mentioned in the issue #90 

## Proposed changes (including videos or screenshots)

- Improved database schema for re-usability of component for different repositories.  ![image](https://user-images.githubusercontent.com/70485812/158068629-ac5f6c64-426f-48aa-849b-e15959f57f16.png)
- Improved data fetching method in cms (can be re-used when switching from strapi) to provide a better experience for community builders. All data fetching for  the github kit for a repository will be done by a a single `githubKit()` function.
- Replacing individual component tags with a `<Github ... >` tag which can be used to fetch different components of the kit depending on props/parameters. `<Github type={'contributors'} githubData={props.githubData} />`
- Improve github helper functions (which fetch data from the cms into our application) into a single `githubKitData(owner ,name ,[..needs])` function which can fetch relevant data based on the components we wish to use on a particular page.
- New Pull Request Component<p align="center" width="100%">
  <img alt="crons-example" src="https://user-images.githubusercontent.com/70485812/158337347-ae75da38-3b3d-4f9f-a816-4f5e0fe0cc66.png"></p>
- New Repository overview component <p align="center" width="100%">
  <img alt="crons-example" src="https://user-images.githubusercontent.com/70485812/158334060-0f006063-3bae-472c-b85b-b32a018a5e1e.png"></p>
- Updated  [Documentation](https://github.com/samad-yar-khan/RC4Community/tree/githubComponentKit/docs/components/github)  . P.S. Refer to the documentation for changes in usability.

samad-yar-khan commented:
@Sing-Li I have made some changes and removed the unnecessary code, it would be great if you could review the pull request.

Sing-Li reviewed (changes requested):
Good clean up work, congrats.  

However, you added back package.json  and some index.js modifications

Please remove package.json  and the index.js modifications.   Take a look at other PRs and see how they use a demo page under `pages` to demonstrate their component.  Thanks.

samad-yar-khan commented:
@Sing-Li I have made the requested changed : 

- Removed the changes to the `index.js` file. 
- Created a demo page for  the `Github Component Kit` which can be accessed on the route `/github` .
- Removed `.package-lock.json` file .
